datumaro/plugins/open_images_format.py

```diff
@@ -337,6 +372,89 @@ def _load_bboxes(self, items_by_id):
                         attributes=attributes,
                     ))
 
+    def _load_masks(self, items_by_id):
+        label_categories = self._categories[AnnotationType.label]
+
+        for mask_path in self._glob_annotations(
+            '*' + OpenImagesPath.MASK_DESCRIPTION_FILE_SUFFIX
+        ):
+            with self._open_csv_annotation(mask_path) as mask_reader:
+                for mask_description in mask_reader:
+                    mask_path = mask_description['MaskPath']
+                    if _RE_INVALID_PATH_COMPONENT.fullmatch(mask_path):
+                        raise UnsupportedMaskPathError(item_id=item.id, mask_path=mask_path)
+
+                    image_id = mask_description['ImageID']
+                    item = items_by_id[image_id]
+
+                    label_name = mask_description['LabelName']
+                    label_index, _ = label_categories.find(label_name)
+                    if label_index is None:
+                        raise UndefinedLabel(
+                            item_id=item.id, subset=item.subset,
+                            label_name=label_name, severity=Severity.error)
+
+                    attributes = {}
+
+                    # The box IDs are rather useless, because the _box_ annotations
+                    # don't include them, so they cannot be used to match masks to boxes.
+                    # However, it is still desirable to record them, because they are
+                    # included in the mask file names, so in order to save each mask to the
+                    # file it was loaded from when saving in-places, we need to know
+                    # the original box ID.
+                    box_id = mask_description['BoxID']
+                    if _RE_INVALID_PATH_COMPONENT.fullmatch(box_id):
+                        raise UnsupportedBoxIdError(item_id=item.id, box_id=box_id)
+                    attributes['box_id'] = box_id
+
+                    # In the original OID, the masks are often bigger than
+                    # the images, which doesn't fit Datumaro's data model.
+                    # We also need the image size to denormalize the box coordinates.
+                    # As a workaround, we resize all masks to the image's size.
+                    if not item.has_image or item.image.size is None:
+                        log.warning(
+                            "Can't load mask for item '%s' due to missing image file",
+                            item.id)
+                        continue
+
+                    height, width = item.image.size
+
+                    # The original OID has box coordinates for all masks, but
+                    # a dataset converted from another dataset might not.
+                    if all(mask_description[f] for f in [
+                        'BoxXMin' ,'BoxXMax', 'BoxYMin', 'BoxYMax',
+                    ]):
+                        box = attributes['box'] = {}
```

Comment:
Is there any reason to include it as an attribute? Can these values be distinct from what is in the mask? If possible, let's stick to simple type attributes.
If you want to have a bbox attached to a mask, the current approach is to have it as another annotation in the list with the same group as the corresponding mask and no label. It complicates processing a bit, though, check COCO converter or https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/util/annotation_util.py#L15. I'd vote for exclusion of this information completely, because it can be computed without great overhead, and masks have such method.

Reply:
> Is there any reason to include it as an attribute? Can these values be distinct from what is in the mask?

To quote the Open Images website:

"Note that this is not the bounding box of the mask, but the starting box from which the mask was annotated. These coordinates can be used to relate the mask data with the boxes data."

> If you want to have a bbox attached to a mask, the current approach is to have it as another annotation in the list with the same group as the corresponding mask and no label.

Okay, that seems quite doable. I can assign a distinct group to every box annotation and then find the matching box annotation for every mask annotation when loading them.

I could, of course, just not load this information at all, but according to the dataset description, it's _not_ something just can be computed, so it seems useful to have.

Reply:
> Okay, that seems quite doable. I can assign a distinct group to every box annotation and then find the matching box annotation for every mask annotation when loading them.

Implemented this.